### PR TITLE
Add Endpoint /testing/history

### LINKED ISSUE #176: Feature Request: Testzahlen
Hallo,
 wäre es machbar, die Gesamtzahl der Tests mit darzustellen?

Ich kenne leider nur die eine Tabelle vom RKI:
[https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Daten/Testzahlen-gesamt.xlsx?__blob=publicationFile](url)

Vielleicht gibt es noch eine weitere. 

Danke.

### PR DESCRIPTION
**Pull request by Rubber1Duck**
add a new endpoint /testing/history witch provides the  number of weekly done PCR-tests, the number of positive test in this week, the positiv tests quote and the number of transmitting laboratories.
Fix #176

### CODE REVIEW

**Comment by Rubber1Duck** [comment]
so, nu aber :-)

testing

**Comment by beyerleinf** [comment]
Hey @Rubber1Duck, super Arbeit! Bei "PositivQuote" und "PositivTesting" fehlt nur ein "e" :) Ich glaube bei den properties sollte der erste Buchstabe auch klein sein :)

**Comment by Rubber1Duck** [comment]
> 
> 
> Hey @Rubber1Duck, super Arbeit! Bei "PositivQuote" und "PositivTesting" fehlt nur ein "e" :) Ich glaube bei den properties sollte der erste Buchstabe auch klein sein :)

Danke für die Tipps! erledigt

**Comment by beyerleinf** [comment]
Looks good!

**Comment by Rubber1Duck** [comment]
@marlon360 merge please. i have ready another endpoint /maps/districts/history/:days witch gives the history districts map up to 30 days in the past! but first this PR must merged!
and also merge PR #174 because of the documentation.
thanks

File: src/data-requests/testing.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import axios from "axios";
+import XLSX from 'xlsx';
+import { ResponseData } from "./response-data";
+
+export interface testingHistoryEntry{
+    calendarWeek: string,
+    countTesting: number,
```
Comment (marlon360) [inline src/data-requests/testing.ts]: Can we change this to `performedTests`?

File: src/data-requests/testing.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    positiveTesting: number,
```
Comment (marlon360) [inline src/data-requests/testing.ts]: Can we change this to `positiveTests`?

File: src/data-requests/testing.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    positiveQuote: number,
```
Comment (marlon360) [inline src/data-requests/testing.ts]: Can we change this to `positivityRate`?

File: src/data-requests/testing.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    countLaboratories: number
```
Comment (marlon360) [inline src/data-requests/testing.ts]: Can we change this to `laboratoryCount`?

File: docs/index.md [hunk 5]
```diff
@@ -33,6 +34,7 @@ If you use this API, please consider supporting me:
 - data per state and district
 - time series for every state and district
 - maps for states and districts
+- Count of the weekly carried out PCR-Tests, Count of positiv tests and positiv quotes
```
Comment (marlon360) [inline docs/index.md]: Better "Number of performed PCR-test, number of positive tests and positivity rate"

File: docs/endpoints/testing.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+### Request
+
+`GET https://api.corona-zahlen.org/testing/history`
+[Open](/testing/history)
+
+### Response
+
+`calendarWeek` CalendarWeek
+
+`countTesting` Number PCR-tests executed.
```
Comment (marlon360) [inline docs/endpoints/testing.md]: "Number of performed PCR-tests"

File: docs/endpoints/testing.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+`positiveTesting` Number PCR-tests witch are COVID-19 positiv
```
Comment (marlon360) [inline docs/endpoints/testing.md]: "Number of positive PCR-tests"

File: docs/endpoints/testing.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+`positiveQuote` Ouote of COVID-19 positive tests
```
Comment (marlon360) [inline docs/endpoints/testing.md]: "The rate of positive PCR-tests"

File: docs/endpoints/testing.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+`countLaboratories` Number Laboratories witch transmitted Data to RKI
```
Comment (marlon360) [inline docs/endpoints/testing.md]: "Number of laboratories which transmitted data to the RKI"

File: src/data-requests/testing.ts [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        } else if (entry.Kalenderwoche === "Summe") {
+        // do nothing, skip this entry
+        }  
+        else { 
+            testingHistory.push({
+            calendarWeek: entry.Kalenderwoche,
+            countTesting: entry['Anzahl Testungen'],
+            positiveTesting: entry['Positiv getestet'],
+            positiveQuote: entry['Positivenanteil (%)'] / 100,
+            countLaboratories: entry['Anzahl übermittelnder Labore']
```
Comment (marlon360) [inline src/data-requests/testing.ts]: Please add a "tab" to each line for better formatting.

File: README.md [hunk 11]
```diff
@@ -33,6 +34,7 @@ If you use this API, please consider supporting me:
 - data per state and district
 - time series for every state and district
 - maps for states and districts
+- Count of the weekly carried out PCR-Tests, Count of positiv tests and positiv quotes
```
Comment (marlon360) [inline README.md]: Better "Number of performed PCR-test, number of positive tests and positivity rate"

**Review by marlon360 — CHANGES_REQUESTED** [review]
Hi @Rubber1Duck!
Thank you for this PR.
I would like to change the naming of some attributes and some Englisch translations.

**Comment by marlon360** [comment]
> @marlon360 merge please. i have ready another endpoint /maps/districts/history/:days witch gives the history districts map up to 30 days in the past! but first this PR must merged!
> and also merge PR #174 because of the documentation.
> thanks

Nice @Rubber1Duck!
Maybe consider using different branches for different PRs, so you can work on them simultaneously.

**Comment by Rubber1Duck** [comment]
> 
> 
> > @marlon360 merge please. i have ready another endpoint /maps/districts/history/:days witch gives the history districts map up to 30 days in the past! but first this PR must merged!
> > and also merge PR #174 because of the documentation.
> > thanks
> 
> Nice @Rubber1Duck!
> Maybe consider using different branches for different PRs, so you can work on them simultaneously.

Hi Marlon, of course i will do the changes!
I'm fighting with github ......... :-) i hope we will get good friends soon, but at the moment i despair sometimes

**Comment by Rubber1Duck** [comment]
changes done

**Comment by marlon360** [comment]
Thanks!
But I think in this PR are now changes that don’t belong here. Could you clean that up?

**Comment by Rubber1Duck** [comment]
New PR #180 !

**Pull request closed without merging**